containers/src/Data/Set.hs

```diff
@@ -70,6 +70,7 @@ module Data.Set (
               Set          -- instance Eq,Ord,Show,Read,Data,Typeable
 #else
               Set(..)
+            , NonEmptySet(..)
```

Comment:
Is there some discussion where and how `NonEmptySet` should be exported. I'd prefer a separate module to make haddocks cleaner.

I have no preference, I guess it's fine that `NonEmptySet` methods have `NE` suffix

Reply:
@phadej, the tentative idea is to rename them in a separate module, taking advantage of mutual recursion internally when doing so makes sense.

Reply:
@treeowl that sounds very good. :+1:

Reply:
Yeah it's only exported there for testing purposes. Perhaps the tests should just import the internals module instead of this CPP too.